okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -480,9 +480,10 @@ int getPendingStreamSize() {
   /**
    * Finish all active streams due to an IOException, then close the transport.
    */
-  void onException(Throwable failureCause) {
-    log.log(Level.WARNING, "Transport failed", failureCause);
-    startGoAway(0, Status.UNAVAILABLE.withCause(failureCause));
+  void onException(Throwable cause) {
+    log.log(Level.WARNING, "Transport failed", cause);
+    startGoAway(0, Status.UNAVAILABLE.withCause(cause)
+        .augmentDescription(cause == null ? "No provided cause" : cause.getMessage()));
```

Comment:
I'm okay with handling the null case by saying `"No provided cause"` or, better in my mind without looking too deeply at it, enforcing `cause` to be non-`null`. I just verified all existing usages of the method (and there are some in `AsyncFrameWriter`) are guaranteed to be non-`null`.